File: plugin/src/main/java/org/godotengine/plugin/vr/oculus/mobile/api/OvrInitConfig.kt
/**
 * Used to access and setup the init parameters for the plugin.
 *
 * The methods defined in this class must be invoked prior to initializing the Oculus mobile plugin.
 */
@file:JvmName("OvrInitConfig")

package org.godotengine.plugin.vr.oculus.mobile.api

import org.godotengine.plugin.vr.oculus.mobile.OvrMobilePlugin

/**
 * Set the factor by which to multiply the recommended render target size for the app.
 */
fun OvrMobilePlugin.setRenderTargetSizeMultiplier(multiplier: Double): Boolean {
    return if (isSharedLibLoaded()) {
        nativeSetRenderTargetSizeMultiplier(multiplier)
    } else {
        false
    }
}

private external fun nativeSetRenderTargetSizeMultiplier(multiplier: Double): Boolean

File: plugin/src/main/cpp/api/ovr_init_config.cpp
#include "ovr_init_config.h"
#include "api_common.h"

namespace ovrmobile {

    bool set_render_target_size_multiplier(OvrMobileSession *session, double multiplier) {
        return check_session_not_initialized<bool>(session, [&](){
            // Update the render target size.
            session->set_render_target_size_multiplier(multiplier);
            return true;
        }, []() { return false; });
    }

} // namespace ovrmobile